[CD][CUDA][Triton][PTXAS] Turn on  BUILD_BUNDLE_PTXAS=1 for CUDA13 X86 Wheel Build

nWEIdia opened the pull request:
Triton (release/3.5.x) by default ships CUDA12.8 ptxas. 
This PR tries to bundle a ptxas version for cuda13, so that it can help #163801 when users run on new devices like THOR and Spark. 

Fixes #163801 

Test Plan: 
1) Check binary size increase against nightly or v2.9RC 
2) Install the binary from into a working THOR and B200/H100 machine (reproduce the original issue first on THOR), then install the binary built from this PR and we expect the issue to be gone without any additional user setting. Testing on B200 is to ensure no regression.  

[Update: since the original ptxas issue seems to be an ARM specific issue - THOR device is on ARM], This PR seems not appropriate. 

Reference: https://github.com/pytorch/pytorch/pull/119750 and https://github.com/pytorch/builder/commit/5c814e2527b3f5797488bf57d9d5425e63dcc1ac 

cc @ptrblck @eqy @tinglvv @atalman @malfet

nWEIdia committed:
[CD][CUDA][Triton][PTXAS] Turn on  BUILD_BUNDLE_PTXAS=1 for CUDA13
builds because triton by default ships CUDA12.8 ptxas

nWEIdia committed:
Undo submodule changes

nWEIdia committed:
Lint fix. sorry.

malfet reviewed (approved):
LGTM, but please update test plan
And propose followup PR for aarch64

nWEIdia commented:
> LGTM, but please update test plan And propose followup PR for aarch64

Thanks @malfet ! I updated the test plan and changed the title to reflect this is only x86. I will create a follow up for ARM CUDA13 wheel build, thanks for catching it!

nWEIdia commented:
Closing as the new ptxas that comes with cuda13 is mostly dealing with ARM based GPU addition. So having a cuda13 ptxas in x86 cuda binary does not have clear help. 
Prioritizing #163988 instead.
